homeassistant/components/intesishome/climate.py

```diff
@@ -0,0 +1,413 @@
+        persistent_notification.create(
+            hass, "Wrong username/password.", device_type, "intesishome"
+        )
+        await controller.stop()
+        raise PlatformNotReady()
+    else:
+        persistent_notification.create(
+            hass, controller.error_message, device_type, "intesishome"
+        )
+        await controller.stop()
+        raise PlatformNotReady()
+
+
+class IntesisAC(ClimateDevice):
+    """Represents an Intesishome air conditioning device."""
+
+    def __init__(self, deviceid, ih_device, controller):
+        """Initialize the thermostat."""
+        _LOGGER.debug("Added climate device with state: %s", repr(ih_device))
+        self._controller = controller
+
+        self._deviceid = deviceid
+        self._devicename = ih_device.get("name")
+        self._devicetype = controller.device_type
+        self._has_swing_control = IH_SWING_WIDGET in ih_device.get("widgets")
+        self._connected = controller.is_connected
+        self._current_temp = None
+        self._setpoint_step = ih_device.get("farenheit_type")
+
+        self._max_temp = None
+        self._min_temp = None
+        self._target_temp = None
+
+        self._run_hours = None
+        self._rssi = None
+        self._swing = None
+        self._swing_list = None
+        self._vvane = None
+        self._hvane = None
+        self._power = False
+        self._fan_speed = STATE_UNKNOWN
+        self._hvac_mode = STATE_UNKNOWN
+        self._connection_retries = 0
+
+        self._hvac_modes = [
+            HVAC_MODE_HEAT_COOL,
+            HVAC_MODE_COOL,
+            HVAC_MODE_HEAT,
+            HVAC_MODE_DRY,
+            HVAC_MODE_FAN_ONLY,
+            HVAC_MODE_OFF,
+        ]
+        self._fan_modes = controller.get_fan_speed_list(deviceid)
+
+        self._support = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+
+        if self._has_swing_control:
+            self._support |= SUPPORT_SWING_MODE
+            self._swing_list = [SWING_OFF, SWING_BOTH, SWING_VERTICAL, SWING_HORIZONTAL]
+
+        self._controller.add_update_callback(self.update_callback)
+
+    @property
+    def name(self):
+        """Return the name of the AC device."""
+        return self._devicename
+
+    @property
+    def temperature_unit(self):
+        """Intesishome API uses celsius on the backend."""
+        return TEMP_CELSIUS
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        attrs = {}
+        if self._has_swing_control:
+            attrs["vertical_vane"] = self._vvane
+            attrs["horizontal_vane"] = self._hvane
+
+        if self._controller.is_connected:
+            attrs["ha_update_type"] = "push"
+        else:
+            attrs["ha_update_type"] = "poll"
+
+        return attrs
+
+    @property
+    def unique_id(self):
+        """Return unique ID for this device."""
+        return self._deviceid
+
+    @property
+    def device_info(self):
```

Comment:
The `device_info` property is only active for entities under an integration supporting config entries. So it won't be active in our case. Please remove it for now.

https://developers.home-assistant.io/docs/en/device_registry_index.html#defining-devices